### PR TITLE
Preparing a YANG model for YANG doctor review

### PR DESCRIPTION
**Pull request by sergiobelotti**
Proposed updates to the YANG model for Path Computation RPC to:
- align with the latest version of the TE tunnel YANG model (https://tools.ietf.org/html/draft-ietf-teas-yang-te-23)
- add support of path computation for protected paths
- add support of path computation for bidirectional paths
- returning errors in case path computation fails
- add support for multi-layer path computation

The objective is to have a YANG model ready for requesting YANG doctor review

### CODE REVIEW

**Comment by italobusi** [comment]
@ju7ien , @dieterbeller , @haomianzheng : Please take a look at section 5.3 of the updated editors' draft to explain most of the major changes proposed with this pull request:

[draft-ietf-teas-yang-path-computation-09@2020-04-20.docx](https://github.com/rvilalta/ietf-te-path-computation/files/4504630/draft-ietf-teas-yang-path-computation-09%402020-04-20.docx)

**Comment by italobusi** [comment]
Updated version of the editors' draft:

[draft-ietf-teas-yang-path-computation-09@2020-05-25.docx](https://github.com/rvilalta/ietf-te-path-computation/files/4678143/draft-ietf-teas-yang-path-computation-09%402020-05-25.docx)

**Comment by italobusi** [comment]
Latest Editors' draft: 
[draft-ietf-teas-yang-path-computation-09@2020-06-04.docx](https://github.com/rvilalta/ietf-te-path-computation/files/4728673/draft-ietf-teas-yang-path-computation-09%402020-06-04.docx)

Sections 5.3 and 5.4 describes the major changes to the model provided in the PR

File: ietf-te-path-computation.yang [hunk 1]
```diff
@@ -497,27 +535,315 @@ module ietf-te-path-computation {
...
     list path-request {
       key "request-id";
-      description "request-list";
+      description "The list of the requested paths to be computed";
       leaf request-id {
         type uint32;
         mandatory true;
         description
           "Each path computation request is uniquely identified
            by the request-id-number.";
```
Comment (dieterbeller) [inline ietf-te-path-computation.yang:551]: The request-id-number seems to be provided by the client. What is the scope in which the request-id-number is unique? Is it unique within a NETCONF session for example?

File: ietf-te-path-computation.yang [hunk 2]
```diff
@@ -25,41 +37,122 @@ module ietf-te-path-computation {
...
+
   /*
-   * Features
+   * Identities
    */
 
-  feature stateless-path-computation {
+  identity svec-metric-type {
+    description
+      "Base identity for svec metric type";
```
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-te-path-computation.yang]: Addressed in https://github.com/rvilalta/ietf-te-path-computation/pull/70/commits/7cf3781791607c49b0795b4a7b892ca99289a6a3

File: ietf-te-path-computation.yang [hunk 3]
```diff
@@ -25,41 +37,122 @@ module ietf-te-path-computation {
...
+  }
+
+  identity svec-metric-cumul-te {
+    base svec-metric-type;
+    description
+      "TE cumulative path metric";
```
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-te-path-computation.yang]: https://github.com/rvilalta/ietf-te-path-computation/pull/70/commits/7cf3781791607c49b0795b4a7b892ca99289a6a3

File: ietf-te-path-computation.yang [hunk 4]
```diff
@@ -25,41 +37,122 @@ module ietf-te-path-computation {
...
+  }
+
+  identity svec-metric-cumul-igp {
+    base svec-metric-type;
+    description
+      "IGP cumulative path metric";
```
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-te-path-computation.yang]: https://github.com/rvilalta/ietf-te-path-computation/pull/70/commits/7cf3781791607c49b0795b4a7b892ca99289a6a3

File: ietf-te-path-computation.yang [hunk 5]
```diff
@@ -25,41 +37,122 @@ module ietf-te-path-computation {
...
-      "This feature indicates that the system supports
-       stateless path computation.";
+      "Hop cumulative path metric";
+  }
+
+  identity svec-metric-aggregate-bandwidth-consumption {
+    base svec-metric-type;
+    description
+      "Cumulative bandwith consumption of the set of
+       synchronized paths";
```
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-te-path-computation.yang]: Addressed in https://github.com/rvilalta/ietf-te-path-computation/pull/70/commits/7cf3781791607c49b0795b4a7b892ca99289a6a3

File: ietf-te-path-computation.yang [hunk 6]
```diff
@@ -25,41 +37,122 @@ module ietf-te-path-computation {
...
+    description
+      "IGP cumulative path metric";
+  }
+
+  identity svec-metric-cumul-hop {
+    base svec-metric-type;
     description
-      "This feature indicates that the system supports
-       stateless path computation.";
+      "Hop cumulative path metric";
```
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-te-path-computation.yang]: See: https://github.com/rvilalta/ietf-te-path-computation/issues/77

File: ietf-te-path-computation.yang [hunk 7]
```diff
@@ -25,41 +37,122 @@ module ietf-te-path-computation {
...
+    base svec-metric-type;
+    description
+      "Cumulative bandwith consumption of the set of
+       synchronized paths";
+  }
+
+  identity svec-metric-load-of-the-most-loaded-link {
+    base svec-metric-type;
+    description
+      "Load of the most loaded link";
```
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-te-path-computation.yang]: Addressed in https://github.com/rvilalta/ietf-te-path-computation/pull/70/commits/7cf3781791607c49b0795b4a7b892ca99289a6a3

File: ietf-te-path-computation.yang [hunk 8]
```diff
@@ -25,41 +37,122 @@ module ietf-te-path-computation {
...
+  }
+
+  identity tunnel-action-path-compute-delete {
+    base te:tunnel-actions-type;
+    description
+      "Action type to delete the transient states
+       of computed paths";
```
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-te-path-computation.yang]: Addressed in https://github.com/rvilalta/ietf-te-path-computation/pull/70/commits/7cf3781791607c49b0795b4a7b892ca99289a6a3

File: ietf-te-path-computation.yang [hunk 9]
```diff
@@ -25,41 +37,122 @@ module ietf-te-path-computation {
...
    */
 
-  grouping path-info {
-    uses te-types:generic-path-properties;
-    description "Path computation output information";
-  }
+  grouping protection-restoration-properties {
+    description
+      "This grouping defines the type of protection and restoration
+       to be used by a requested path";
```
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-te-path-computation.yang]: Addressed in https://github.com/rvilalta/ietf-te-path-computation/pull/70/commits/7cf3781791607c49b0795b4a7b892ca99289a6a3

File: ietf-te-path-computation.yang [hunk 10]
```diff
@@ -25,41 +37,122 @@ module ietf-te-path-computation {
...
+      }
+      default te-types:lsp-protection-unprotected;
+      description "LSP protection type.";
+    }
+    leaf restoration-type {
+      type identityref {
+        base te-types:lsp-restoration-type;
+      }
+      default te-types:lsp-restoration-restore-any;
+      description "LSP restoration type.";
```
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-te-path-computation.yang:167]: The default is defined in te-types

File: ietf-te-path-computation.yang [hunk 11]
```diff
@@ -94,7 +187,7 @@ module ietf-te-path-computation {
        If false, returning path affinities in the response is
        optional.";
     }
-  }
+  }   // grouping requested-info
 
   grouping requested-state {
     description
```
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-te-path-computation.yang:208]: Addressed in https://github.com/rvilalta/ietf-te-path-computation/pull/70/commits/7cf3781791607c49b0795b4a7b892ca99289a6a3

File: ietf-te-path-computation.yang [hunk 12]
```diff
@@ -130,61 +223,7 @@ module ietf-te-path-computation {
...
-             by the server.
-            ";
-        }
-      }
-    }
-  }
+  }   // grouping requested-state
 
   grouping reported-state {
     description
```
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-te-path-computation.yang:242]: Addressed in https://github.com/rvilalta/ietf-te-path-computation/pull/70/commits/7cf3781791607c49b0795b4a7b892ca99289a6a3

File: ietf-te-path-computation.yang [hunk 13]
```diff
@@ -204,93 +243,102 @@ module ietf-te-path-computation {
...
+              "The primary-reverse-path name can only be reported
+               if also the tunnel name is reported.";
+          }
+          description
+            "
+             Reference to the primary-reverse-path that reports
+             the transient state of the computed path.
+
+             If no transient state is created,
+             this attribute is empty.
```
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-te-path-computation.yang]: Addressed in https://github.com/rvilalta/ietf-te-path-computation/pull/70/commits/7cf3781791607c49b0795b4a7b892ca99289a6a3

File: ietf-te-path-computation.yang [hunk 14]
```diff
@@ -204,93 +243,102 @@ module ietf-te-path-computation {
...
-  }
+             If no transient state is created,
+             this attribute is empty.
+            ";
+        }
+      } // case secondary
+    }   // choice path
+  }   // grouping reported-state
 
   grouping svec-metrics-bounds_config {
```
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-te-path-computation.yang]: Addressed in https://github.com/rvilalta/ietf-te-path-computation/pull/70/commits/7cf3781791607c49b0795b4a7b892ca99289a6a3

File: ietf-te-path-computation.yang [hunk 15]
```diff
@@ -204,93 +243,102 @@ module ietf-te-path-computation {
...
     description
```
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-te-path-computation.yang:359]: Addressed in https://github.com/rvilalta/ietf-te-path-computation/pull/70/commits/7cf3781791607c49b0795b4a7b892ca99289a6a3

File: ietf-te-path-computation.yang [hunk 16]
```diff
@@ -307,7 +355,7 @@ module ietf-te-path-computation {
       type uint64;
       description "Upper bound on end-to-end svec path metric";
     }
-  }
+  }   // grouping svec-metrics-bounds_config
 
   grouping svec-metrics-optimization_config {
```
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-te-path-computation.yang]: Addressed in https://github.com/rvilalta/ietf-te-path-computation/pull/70/commits/7cf3781791607c49b0795b4a7b892ca99289a6a3

File: ietf-te-path-computation.yang [hunk 17]
```diff
@@ -360,7 +408,7 @@ module ietf-te-path-computation {
     }
     uses te-types:generic-path-srlgs;
     uses svec-exclude;
-  }
+  }   // grouping synchronization-constraints
 
   grouping synchronization-optimization {
       description "Synchronized request optimization";
```
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-te-path-computation.yang]: Addressed in https://github.com/rvilalta/ietf-te-path-computation/pull/70/commits/7cf3781791607c49b0795b4a7b892ca99289a6a3

File: ietf-te-path-computation.yang (hunk 1 above)
Comment (italobusi) [reply ietf-te-path-computation.yang]: Addressed in https://github.com/rvilalta/ietf-te-path-computation/pull/70/commits/7cf3781791607c49b0795b4a7b892ca99289a6a3

**Pull request closed without merging**